web/src/components/CustomMarquee/CustomMarquee.jsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import Marquee from "react-fast-marquee";
+
+import PropTypes from "prop-types";
+
+function CustomMarquee(props) {
+    const {
+        data,
+        containerRef,
+    } = props;
+
+    const windowSize = React.useRef(window.innerWidth);
+    const resizeCooldown = React.useRef(false);
+
+    const dataRef = React.useRef(null);
+    const [dataScroll, setDataScroll] = React.useState(false);
+    const dataOverrideTimeout = React.useRef(null);
+    const dataScrollOverride = React.useRef(true); // set to true by defualt so that there is no scrolling on load, until a setTimeout later on changes that 2 seconds after load
+
+    React.useEffect(() => {assessMarquee();}, [data]);
+
+    function assessMarquee() {
+        const container = containerRef.current;
+        const scroll = dataRef.current;
+
+        if(container && scroll){
+            if(container.offsetWidth < scroll.scrollWidth && !dataScrollOverride.current){
+                setDataScroll(true);
+            } else if(container.offsetWidth >= scroll.scrollWidth && dataScrollOverride.current){ // dataScroll is only able to be set to false when a scroll has been completed to avoid randomly jumbled text
+                clearTimeout(dataOverrideTimeout.current);
+            }
+        }
+    }
+
+    function setOverride(){
+        setDataScroll(false);
+        dataScrollOverride.current = true;
+
+        dataOverrideTimeout.current = setTimeout(()=>{dataScrollOverride.current = false; assessMarquee();}, 2000); // When a scroll completes, pause for two seconds and restart (unless the timeout is cleared)
+    }
+
+    setTimeout(()=>{dataScrollOverride.current = false; assessMarquee();}, 2000) // used to kick off useEffect after two seconds of page load
+
+
+    function handleResize(){
+        if(!resizeCooldown.current && window.innerWidth != windowSize.current){
```

Comment:
is the `innerWidth != windowSize.current` necessary? feels like you kinda want to fire on every resize where cooldown isn't doing the thing; probably will get rid of some extra state too